### PR TITLE
Remove TrackInteractions & COMPONENT_INTERACTION_TRACKED

### LINKED ISSUE #11726: TrackInteractions: Drop react-redux private API import
[This line](https://github.com/Automattic/wp-calypso/blob/854c7104d13df6c99a2b33a14c93d5d379a6dadb/client/components/track-interactions/index.js#L4) [prevents us](https://github.com/reactjs/react-redux/issues/632) from upgrading to `react-redux` 5.0.3. 😕 We should get rid of it.

PSA: Please no private API imports, ever. I've spent way too much time on getting rid of them in preparation for React upgrades.

### PR DESCRIPTION
**Pull request by mcsf**
Fixes #11726 

See #11739 for motive.

# Testing

Everything should be unaffected, including `Theme` and `/design`. The only affected piece would have been `DesignShowcaseWelcomeTour`, which is currently disabled.

### CODE REVIEW

File: client/state/ui/guided-tours/contexts.js [hunk 1]
```diff
@@ -85,16 +85,15 @@ export const hasUserRegisteredBefore = date => state => {
...
+export const hasUserInteractedWithComponent = () => () => false;
+
 /**
- * Returns a selector that tests whether the user has interacted with a given component.
- *
- * @see client/components/track-interactions
  *
- * @param {String} componentName Name of component to test
  * @return {Function} Selector function
  */
```
Comment (ockham) [inline client/state/ui/guided-tours/contexts.js]: This JSDoc block seems obsolete -- no function below it AFAICS.

Comment (mcsf) [inline client/state/ui/guided-tours/contexts.js]: Ah, that was some bad partial Git staging, my fault. 😅

File: client/state/ui/action-log/selectors.js [hunk 2]
```diff
@@ -30,6 +30,6 @@ export function getActionLog( state ) {
  * @return {Object}              The matching dispatched action
  */
 export const getLastAction = createSelector(
-	( state ) => last( state.ui.actionLog ),
+	( state ) => last( state.ui.actionLog ) || false,
```
Comment (ockham) [inline client/state/ui/action-log/selectors.js]: This is throwing an error now (see CircleCI), so it seems like you'll need to modify a test too 😄.

File: client/state/ui/guided-tours/contexts.js (hunk 1 above)
Comment (ockham) [inline client/state/ui/guided-tours/contexts.js]: Happens to the best 😎

**Comment by mtias** [comment]
Always nice to see a bunch of red.

**Pull request merged**